### ISSUE TITLE
Can't reload quantized MobileNetV3 - quantization parameters of QuantizedHardSwish don't get saved

### ISSUE THREAD

**Issue by guybuk** [pytorch/pytorch#91877]
## Issue description
When attempting to save and reload a quantized MobileNetV3 model, the observed `scale` and `zero_point` parameters of the `QuantizedHardSwish` layer are not saved, and thus cannot be reloaded properly (leading to a broken model). This is likely due to the fact that, unlike `torch.ao.nn.quantized.modules.conv.Conv2d`, `torch.ao.nn.quantized.modules.activation.Hardswish` does not override `_save_to_state_dict` and `_load_from_state_dict` to include the `scale` and `zero_point` parameters in the state_dict and load them afterwards.

## Code example
```
import torch
import io
from torchvision.models.mobilenetv3 import mobilenet_v3_large
from torch.ao.quantization import quantize_fx

# code similar to the example given in https://pytorch.org/docs/stable/quantization.html#saving-and-loading-quantized-models

model1 = mobilenet_v3_large(weights='DEFAULT').eval()

model_to_quantize=model1.eval()
model_prepared = quantize_fx.prepare_qat_fx(model_to_quantize, torch.ao.quantization.get_default_qat_qconfig_mapping('fbgemm'), torch.randn(1,3,224,224))
[model_prepared(torch.randn(1,3,224,224)) for _ in range(5)] # calibrate
model_quantized_orig = quantize_fx.convert_fx(model_prepared)

b = io.BytesIO()
torch.save(model_quantized_orig.state_dict(), b)

model2 = mobilenet_v3_large(weights='DEFAULT').eval()
model_to_quantize=model2.eval()
model_prepared = quantize_fx.prepare_qat_fx(model_to_quantize, torch.ao.quantization.get_default_qat_qconfig_mapping('fbgemm'),torch.randn(1,3,224,224))
model_prepared.to("cpu")
model_quantized_new = quantize_fx.convert_fx(model_prepared)
b.seek(0)
loaded = torch.load(b)
m = model_quantized_new.load_state_dict(loaded)
print(m)
print(model_quantized_orig.get_submodule('features.0.2').scale,model_quantized_new.get_submodule('features.0.2').scale) # 'features.0.2' is a QuantizedHardSwish layer
```
Outputs:
```
<All keys matched successfully>
0.10166791826486588 1.0
```
Please try to provide a minimal example to repro the bug.
Error messages and stack traces are also helpful.

## System Info
```
Collecting environment information...
PyTorch version: 1.13.0
Is debug build: False
CUDA used to build PyTorch: 11.7
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.5 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.31

Python version: 3.10.8 | packaged by conda-forge | (main, Nov 22 2022, 08:23:14) [GCC 10.4.0] (64-bit runtime)
Python platform: Linux-5.15.79.1-microsoft-standard-WSL2-x86_64-with-glibc2.31
Is CUDA available: True
CUDA runtime version: 11.7.99
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 3070 Laptop GPU
Nvidia driver version: 526.98
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] efficientnet-pytorch==0.7.1
[pip3] mypy==0.991
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.23.5
[pip3] pytorch-lightning==1.7.7
[pip3] segmentation-models-pytorch==0.3.0
[pip3] torch==1.13.0
[pip3] torchmetrics==0.10.3
[pip3] torchvision==0.14.0
[conda] blas                      2.116                       mkl    conda-forge
[conda] blas-devel                3.9.0            16_linux64_mkl    conda-forge
[conda] captum                    0.5.0                         0    pytorch
[conda] efficientnet-pytorch      0.7.1              pyhd8ed1ab_1    conda-forge
[conda] libblas                   3.9.0            16_linux64_mkl    conda-forge
[conda] libcblas                  3.9.0            16_linux64_mkl    conda-forge
[conda] liblapack                 3.9.0            16_linux64_mkl    conda-forge
[conda] liblapacke                3.9.0            16_linux64_mkl    conda-forge
[conda] mkl                       2022.1.0           h84fe81f_915    conda-forge
[conda] mkl-devel                 2022.1.0           ha770c72_916    conda-forge
[conda] mkl-include               2022.1.0           h84fe81f_915    conda-forge
[conda] numpy                     1.23.5          py310h53a5b5f_0    conda-forge
[conda] pytorch                   1.13.0          py3.10_cuda11.7_cudnn8.5.0_0    pytorch
[conda] pytorch-cuda              11.7                 h67b0de4_0    pytorch
[conda] pytorch-lightning         1.7.7              pyhd8ed1ab_0    conda-forge
[conda] pytorch-mutex             1.0                        cuda    pytorch
[conda] segmentation-models-pytorch 0.3.0              pyhd8ed1ab_0    conda-forge
[conda] torchmetrics              0.10.3             pyhd8ed1ab_0    conda-forge
[conda] torchvision               0.14.0              py310_cu117    pytorch
```

**Comment by guybuk** [comment]
Hi. I'm seeing a bit of activity here and there by @fzhao3, but so far there have been no comments or confirmations of actually observing/reproducing this issue. Is there anything still left to do on my end? Is there a timeline/priority for this issue? Thanks.

**Comment by mingfeima** [comment]
@leslie-fang-intel could you please take a look at this issue ?

**Comment by leslie-fang-intel** [comment]
@guybuk I can reproduce this issue and PR: https://github.com/pytorch/pytorch/pull/94486 should fix it. With this PR and running your example code, I get the result of 
```
<All keys matched successfully>
0.10981766134500504 0.10981766134500504
```

**Comment by guybuk** [comment]
@leslie-fang-intel thank you. It may be worth mentioning that the entire module `torch.ao.nn.quantized.modules.activation.py` has activations just like QuantizedHardSwish that may need this fix, such as ELU, Softmax, and PReLU. To my un-educated eyes, the best way to deal with this is how it is dealt in LeakyReLU (in this same module), which is simply to replace:

```
self.scale = scale
self.zero_point=zero_point
```
with:
```
self.register_buffer('scale',scale)
self.register_buffer('zero_point',zero_point)
```

**Comment by leslie-fang-intel** [comment]
@guybuk Yes, the way `LeakyReLU` does should also support the `serialization`. As for the other modules you mentioned (ELU, Softmax, and PReLU), are you interested to provide similar fix? Otherwise, I can also fix it later.
